implementations/kmp-json-schema-validator/src/main/kotlin/BowtieSampsonSchemaValidatorLauncher.kt

```diff
@@ -0,0 +1,214 @@
+
+    init {
+        val attributes: Attributes = javaClass.getResourceAsStream("META-INF/MANIFEST.MF").use {
+            requireNotNull(it) { "cannot find manifest file" }
+            Manifest(it).mainAttributes
+        }
+        with(attributes) {
+            libraryName = getValue("Implementation-Name")
+            libraryVersion = getValue("Implementation-Version")
+            libraryHomepage = getValue("Implementation-Homepage")
+            libraryIssues = getValue("Implementation-Issues")
+        }
+    }
+
+    fun handle(command: Command): Result {
+        return when (command) {
+            is Command.Start -> handleStart(command)
+            is Command.Dialect -> handleDialect(command)
+            is Command.Run -> {
+                handleRun(command)
+                Result.CONTINUE
+            }
+
+            Command.Stop -> Result.STOP
+        }
+    }
+
+    private fun shouldSkipCase(caseDescription: String): String? {
+        return when {
+            caseDescription.endsWith(" format") -> "impl does not support format assertion"
+            caseDescription in IGNORED_CASES || caseDescription.contains("remote ref") ->
+                "impl does not support remote schema loading yet"
+            else -> null
+        }
+    }
+
+    private fun shouldSkipTest(caseDescription: String, testDescription: String): Boolean {
+        // TODO: check tests to skip
+        return false
+    }
+
+    private fun handleRun(command: Command.Run) {
+        requireStarted()
+        requireDialect()
+        shouldSkipCase(command.case.description)?.also { reason ->
+            writer.writeLine(
+                json.encodeToString(
+                    RunResponse.Skipped(
+                        seq = command.seq,
+                        message = "case ${command.case.description} in the list of ignored cases: $reason"
+                    )
+                )
+            )
+            return
+        }
+        val schemaDefinition = command.case.schema
+        // TODO: ability to specify certain dialect
+        val schema = try {
+            JsonSchema.fromJsonElement(schemaDefinition)
+        } catch (ex: Exception) {
+            writer.writeLine(
+                json.encodeToString(
+                    RunResponse.ExecutionError(
+                        seq = command.seq,
+                        context = ErrorContext(
+                            message = "cannot create JSON schema",
+                            traceback = ex.stackTraceToString(),
+                        ),
+                    ),
+                ),
+            )
+            return
+        }
+        val results: List<TestResult> = command.case.tests.map { test ->
+            runCatching {
+                if (shouldSkipTest(command.case.description, test.description)) {
+                    TestResult.Skipped(
+                        message = "test ${test.description} in the list of ignored tests",
```

Comment:
This message is really meant for someone trying to understand why the test is skipped, so it'd be a bit better to either link to an issue in your repository which explains when the functionality might be expected (or otherwise to just say "ref resolution is still being developed" or something).

Reply:
There are no tests skipped at the moment but I will use the same mechanism here as I used for skipping cases